middleware/lib/claim-client.js

```diff
@@ -23,6 +24,16 @@ const { FAMILY_NAME, FAMILY_NAMESPACE, calculateAddress } = createFamily(
 const createTransaction = utils.makeTransactionCreator(FAMILY_NAME, '1.0');
 
 class ClaimClient {
+  /**
+   * Calculate address from a given VIN.
+   *
+   * @param {string} vin - The VIN from which to calculate the address.
+   * @return {string}
+   */
+  static address(vin) {
+    return calculateAddress(vin.replace(/[^A-Za-z0-9]/g, '').toUpperCase());
```

Comment:
Trims out invalid characters (even though this should still be validated in the frontend) and normalizes it to uppercase to prevent duplicate vins being allowed through.